tests/unit/core/test_withExperiment.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable react/no-multi-comp, react/prop-types */
+
+import { shallow } from 'enzyme';
+import * as React from 'react';
+
+import { withExperiment } from 'core/withExperiment';
+import { fakeCookie } from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  describe('withExperiment', () => {
+    class SomeComponentBase extends React.Component {
+      render() {
+        return <div className="component" />;
+      }
+    }
+
+    function componentWithExperiment({ experimentProps, props } = {}) {
+      const TestComponent = withExperiment({
+        id: 'ABtest',
+        variantA: 'AName',
+        variantB: 'BName',
+        ...experimentProps,
+      })(SomeComponentBase);
+
+      return shallow(<TestComponent {...props} />);
```

Comment:
Why not using `shallowUntilTarget()` and look for `SomeComponentBase`?

Reply:
I don't see the variant prop when I use shallowUntilTarget:

```javascript
const root = componentWithExperiment();
const componentBase = root.find('.component');

// this is ok
expect(componentBase).toHaveLength(1);

// this one fails
expect(componentBase).toHaveProp('variant');
```
am I missing something?

Reply:
What if you find `SomeComponentBase` instead?

Reply:
I tried that too and still doesn't work:

```javascript
const root = componentWithExperiment();
const componentBase = root.find(SomeComponentBase);

// this fails (says it should be 0)
expect(componentBase).toHaveLength(1);
...
```

does that look right? Maybe I need to adjust something in the setup function..?